### PR TITLE
Don't assume that WaitOnAddress is not statically available on ARM/ARM64

### LINKED ISSUE #2856: P1135R6_atomic_wait_vista test fails on ARM64 w/ native compiler
With the ARM64 hosted MSVC the test P1135R6_atomic_wait_vista fails on ARM64. It may be making assumptions specific to the intel memory model.

This needs to be investigated.

### PR DESCRIPTION
**Pull request by AlexGuteniev**
Fixes #2856

### CODE REVIEW

File: tests/std/tests/P1135R6_atomic_wait_vista/test.cpp [hunk 1]
```diff
@@ -4,6 +4,11 @@
 #include "test_atomic_wait.hpp"
 
 int main() {
-    assert(__std_atomic_set_api_level(__std_atomic_api_level::__has_srwlock) == __std_atomic_api_level::__has_srwlock);
+    auto level = __std_atomic_set_api_level(__std_atomic_api_level::__has_srwlock);
+#if _ATOMIC_WAIT_ON_ADDRESS_STATICALLY_AVAILABLE
+    assert(level == __std_atomic_api_level::__has_wait_on_address);
+#else // ^^^ wait on address statically available ^^^ / vvv wait on address NOT statically available vvv
+    assert(level == __std_atomic_api_level::__has_srwlock);
+#endif // ^^^ wait on address NOT statically available ^^^
```
Comment (CaseyCarter) [inline tests/std/tests/P1135R6_atomic_wait_vista/test.cpp:12]: 1. If `__std_atomic_set_api_level` breaks this test will now continue to pass, tricking us into thinking we have coverage of the fallback when we don't.
2. This test is redundant with `P1136_atomic_wait` when `assert(level == __std_atomic_api_level::__has_wait_on_address);`.

I think I'd rather see the body of main wrapped in something like `#if !(defined(_M_ARM) || defined(_M_ARM64) || defined(_M_ARM64EC))`.

**Review by CaseyCarter — CHANGES_REQUESTED** [review]

File: tests/std/tests/P1135R6_atomic_wait_vista/test.cpp (hunk 1 above)
Comment (AlexGuteniev) [reply tests/std/tests/P1135R6_atomic_wait_vista/test.cpp:12]: Maybe check for x86 and x64 then instead? Future CPUs will have WaitOnAddress statically available too.

Comment (CaseyCarter) [reply tests/std/tests/P1135R6_atomic_wait_vista/test.cpp]: > Maybe check for x86 and x64 then instead? Future CPUs will have WaitOnAddress statically available too.

Yes, that does make more sense.

**Review by CaseyCarter — APPROVED** [review]

**Comment by barcharcraz** [comment]
I'm running this on the arm64 machine (just as soon as the toolset update installs)

**Comment by barcharcraz** [comment]
This appears to fix the issue, thank you Alex!

**Review by barcharcraz — APPROVED** [review]

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for fixing this ARM64 test failure! :mechanical_arm: :tada: :smiley_cat:

**Pull request merged**